components/raftstore/src/coprocessor/dispatcher.rs

```diff
@@ -674,6 +711,28 @@ mod tests {
             self.called.fetch_add(10, Ordering::SeqCst);
             ctx.bypass = self.bypass.load(Ordering::SeqCst);
         }
+
+        fn pre_apply_snapshot(
+            &self,
+            ctx: &mut ObserverContext<'_>,
+            _: u64,
+            _: &SnapKey,
+            _: Option<&Snapshot>,
+        ) {
+            self.called.fetch_add(17, Ordering::SeqCst);
```

Comment:
any comments for this odd 17 number?

Reply:
There is an unique number for each observer in this mock `TestCoprocessor`. 15/16 is for pre(post)_exec_query(admin) in other PRs.